proposals/vector_to_random_points.json

```diff
@@ -0,0 +1,85 @@
+{
+    "id": "vector_to_random_points",
+    "summary": "Sample random points from geometries",
+    "description": "Generate a vector data cube of points by sampling random points from input geometries.\n\nThrows a `CountsMissing` exception if `geometry_count` and `total_count` are both unrestricted (i.e. set to `null`).",
+    "categories": [
+        "cubes",
+        "vector"
+    ],
+    "experimental": true,
+    "parameters": [
+        {
+            "name": "data",
+            "description": "Input geometries for sample extraction. Vector properties are preserved for vector data cubes and all GeoJSON Features.\n\nTo maximize interoperability, a nested `GeometryCollection` should be avoided. Furthermore, a `GeometryCollection` composed of a single type of geometries should be avoided in favour of the corresponding multi-part type (e.g. `MultiPolygon`).",
+            "schema": [
+                {
+                    "type": "object",
+                    "subtype": "geojson"
+                },
+                {
+                    "type": "object",
+                    "subtype": "vector-cube"
+                }
```

Comment:
Yes. but that's not backward-compatible.

Reply:
> but that's not backward-compatible

yeah, we can not simplify the existing processes, but I was mainly referring to this geojson copy-pasta in all these new vector processes under proposal

Reply:
Not sure. It feels like this is a nice shortcut from a user POV and I'd personally enjoy still having that possibility.

Reply:
Indeed, but I would move the responsibility of accepting and adapting GeoJSON to the client side, and keep the API lean and orthogonal

Reply:
So the client would implicitly also allow geojson always for those types and add a new process load_geometry (or however it's called) to the process graph if present? That seems rather unintuitive and an explicit approach seems better IMHO.

Reply:
Yes that's indeed the idea. What we for example try to do in the Python client is that the user can use Python specific objects and classes (e.g. shapely geometries or numpy/xarray arrays) and the client takes care to convert these things appropriately to openEO process graph JSON format. 
To me that is more intuitive (or at least user friendly) than requiring the user to do all kinds of conversions themselves.

Reply:
Having geojson as an additional data type doesn't restrict you from implementing the additional behavior in a client though. On the other hand, I'll run into a bit of a problem in the JS / Web Editor with this implicit behavior (and I feel it's inconsistent with existing processes).

Reply:
Yes, in client libs we can provide this convenience functionality anyway.

I'm just worried about about the duplication of all the GeoJSON details in all these vector processes:
- what GeoJSON types are supported?
- what GeoJSON types are recommended or to be avoided?
- how are the different GeoJSON types converted to a vector cube: e.g. MultiPolygon becomes single "feature", a GeometryCollection too? Are there any auto-generated feature ids?


That said, if we have some kind of `geojson_to_vectorcube` process, we can also say "see `geojson_to_vectorcube` for conversion rules.

> I'll run into a bit of a problem in the JS / Web Editor with this implicit behavior

I'm not sure what you mean here. A process graph generated by the python client with this implicit conversion will still be a valid process graph, so the web editor will not notice that there was any implicit conversion in play. Or do you mean on how to support implicit conversion in JS/web editor?

> I feel it's inconsistent with existing processes

yes, but that's because in the existing processes there was no alternative to supporting GeoJSON, as vector cubes were practically undefined

Reply:
> I'm just worried about about the duplication of all the GeoJSON details in all these vector processes:
> 
> * what GeoJSON types are supported?

Well, if we add vector cubes it will change to "what geometry types are supported" and be more general. We will still need to make the type distinctions for vector cubes as often you have different behavior based on the type.

> * what GeoJSON types are recommended or to be avoided?
> * how are the different GeoJSON types converted to a vector cube: e.g. MultiPolygon becomes single "feature", a GeometryCollection too? Are there any auto-generated feature ids?

We can basically just say: We allow for GeoJSON what is supported for vector cubes and link to the corresponding docs.

> That said, if we have some kind of `geojson_to_vectorcube` process, we can also say "see `geojson_to_vectorcube` for conversion rules.

We can still have it and link to it. We can basically just say: We allow for GeoJSON what is supported for vector cubes.

> Or do you mean on how to support implicit conversion in JS/web editor?

Yes.

Reply:
Added a separate issue: https://github.com/Open-EO/openeo-processes/issues/343